plugins/aws/firehose/firehoseEncrypted.spec.js

```diff
@@ -0,0 +1,289 @@
+                    err: streamsErr,
+                    data: streams
+                },
+            },
+            describeDeliveryStream: {
+                'us-east-1': {
+                    [stream]: {
+                        data: describeDeliveryStream,
+                        err: describeDeliveryStreamErr
+                    }
+                }
+            }
+        },
+    };
+};
+
+describe('firehoseEncrypted', function () {
+    describe('run', function () {
+        it('should PASS if Firehose delivery stream uses a KMS key for SSE', function (done) {
+            const cache = createCache([listDeliveryStreams[0]], describeDeliveryStream[1]);
+            firehoseEncrypted.run(cache, {}, (err, results) => {
+                expect(results.length).to.equal(1);
+                expect(results[0].status).to.equal(0);
+                expect(results[0].message).to.include('The Firehose delivery stream uses a KMS key for SSE');
+                expect(results[0].region).to.equal('us-east-1');
+                done();
+            });
+        });
+
+        it('should FAIL if Firehose delivery stream does not use a KMS key for SSE', function (done) {
+            const cache = createCache([listDeliveryStreams[0]], describeDeliveryStream[0]);
+            firehoseEncrypted.run(cache, {}, (err, results) => {
+                expect(results.length).to.equal(1);
+                expect(results[0].status).to.equal(2);
+                expect(results[0].message).to.include('The Firehose delivery stream does not use a KMS key for SSE');
+                expect(results[0].region).to.equal('us-east-1');
+                done();
+            });
+        });
+
+ 
+        it('should PASS if No Firehose delivery streams found', function (done) {
+            const cache = createCache([]);
+            firehoseEncrypted.run(cache, {}, (err, results) => {
+                expect(results.length).to.equal(1);
+                expect(results[0].status).to.equal(0);
+                expect(results[0].message).to.include('No Firehose delivery streams found');
+                expect(results[0].region).to.equal('us-east-1');
+                done();
+            });
+        });
+
+        it('should UNKNOWN if Unable to query for firehose delivery streams', function (done) {
+            const cache = createCache(null, null, { message: "Unable to Unable to query Firehose for delivery streams" });
+            firehoseEncrypted.run(cache, {}, (err, results) => {
+                expect(results.length).to.equal(1);
+                expect(results[0].status).to.equal(3);
+                expect(results[0].region).to.equal('us-east-1');
+                done();
+            });
+        });
+
+        it('should UNKNOWN if unable to query Firehose for delivery streams', function (done) {
+            const cache = createCache([listDeliveryStreams[0]], null, { message: "Unable to Unable to query Firehose for delivery streams" });
```

Comment:
what is this test case for, as the above one also having same set of params to createcache? is this for describestreamerr, if so then fix the params order in the createcache call.